src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/DesignTimeCSharpRenderer.cs

```diff
@@ -278,9 +276,15 @@ public override void VisitDeclareTagHelperFields(DeclareTagHelperFieldsIRNode no
 
         private void AddLineMappingFor(RazorIRNode node)
         {
-            var sourceLocation = node.Source.Value;
-            var generatedLocation = new SourceSpan(Context.Writer.GetCurrentSourceLocation(), sourceLocation.Length);
-            var lineMapping = new LineMapping(sourceLocation, generatedLocation);
+            if (node.Source == null)
+            {
+                return;
+            }
+
+            var source = node.Source.Value;
+            
+            var generatedLocation = new SourceSpan(Context.Writer.GetCurrentSourceLocation(), source.Length);
+            var lineMapping = new LineMapping(source, generatedLocation);
```

Comment:
The callers don't consistently check whether or not they have a `.source` which would make this throw if handed a user-synthesized token